packages/frontend/src/main/lib/viewer/commit-object-viewer/stateManager.ts

```diff
@@ -201,301 +211,195 @@ export function setSelectedCommentMetaData(
+ */
+export function getLocalFilterState(): LocalFilterState {
   const state = { ...commitObjectViewerState() }
-  state.hideKey = null
-  state.hideValues = []
-  if (state.isolateKey !== filterKey) {
-    state.isolateValues = []
-  }
-
-  state.isolateKey = filterKey
-  state.isolateValues = state.isolateValues.filter(
-    (val) => filterValues.indexOf(val) === -1
-  )
-  if (state.isolateValues.length === 0) {
-    state.appliedFilter = null
-  } else {
-    state.appliedFilter = {
-      filterBy: { [filterKey]: { includes: state.isolateValues } },
-      ghostOthers: true
-    }
+  const fs = {} as LocalFilterState
+  fs.hiddenIds = state.currentFilterState?.hiddenObjects
+  fs.isolatedIds = state.currentFilterState?.isolatedObjects
+  fs.propertyInfoKey = state.currentFilterState?.activePropFilterKey
+  fs.passMax = state.currentFilterState?.passMax
+  fs.passMin = state.currentFilterState?.passMin
+  const box = getInitializedViewer().getCurrentSectionBox()
+  if (box) {
+    fs.sectionBox = [
+      +box.min.x.toFixed(2),
+      +box.min.y.toFixed(2),
+      +box.min.z.toFixed(2),
+      +box.max.x.toFixed(2),
+      +box.max.y.toFixed(2),
+      +box.max.z.toFixed(2)
+    ]
   }
-
-  updateState(state)
-  getInitializedViewer().applyFilter(state.appliedFilter)
+  return fs
 }
 
-export function hideObjects(params: FilterKeyAndValues) {
-  const { filterKey, filterValues } = params
-  const state = { ...commitObjectViewerState() }
+export function setSectionBox(
+  box?: {
+    min: { x: number; y: number; z: number }
+    max: { x: number; y: number; z: number }
+  },
+  offset?: number
+) {
+  getInitializedViewer().setSectionBox(box, offset)
+}
 
-  state.isolateKey = null
-  state.isolateValues = []
-  if (state.hideKey !== filterKey) {
-    state.hideValues = []
-  }
+export function setSectionBoxFromObjects(objectIds: string[], offset?: number) {
+  getInitializedViewer().setSectionBoxFromObjects(objectIds, offset)
+}
 
-  state.hideKey = filterKey
-  state.hideValues = [...new Set([...filterValues, ...state.hideValues])]
+export function toggleSectionBox() {
+  getInitializedViewer().toggleSectionBox()
+  updateState({ sectionBox: getInitializedViewer().getCurrentSectionBox() !== null })
+}
 
-  if (state.hideValues.length === 0) {
-    state.appliedFilter = null
-  } else {
-    state.appliedFilter = {
-      filterBy: { [filterKey]: { excludes: state.hideValues } }
-    }
-  }
+export function sectionBoxOff() {
+  getInitializedViewer().sectionBoxOff()
+  updateState({ sectionBox: false })
+}
 
-  updateState(state)
-  getInitializedViewer().applyFilter(state.appliedFilter)
+export function sectionBoxOn() {
+  getInitializedViewer().sectionBoxOn()
+  updateState({ sectionBox: true })
 }
 
-export function showObjects(params: FilterKeyAndValues) {
-  const { filterKey, filterValues } = params
-  const state = { ...commitObjectViewerState() }
+export function getObjectProperties() {
```

Comment:
misleading name, the 'get' prefix implies that this function only retrieves some data and doesn't mutate anything. a more appropriate name would be 'loadObjectProperties'